tests/lint/readme.md

```diff
@@ -4,6 +4,72 @@ Our linting process with Flake8 consists of two main steps:
 - Generating a diff
 - Running Flake8 on the diff
 
+## Common problems
+There seem to be several common situations for which the Flake8 errors don't clearly indicate an error free solution.
+
+### Line-breaking functions and statements
+
+Lint errors you may encounter:
+- ET128 (flake8-tabs) *unexpected number of tabs and spaces at start of expression line*
+  - In particular, if this error message "expects x spaces", its likely that it is expecting a
+  code alignment style, rather than hanging indent style.
+  To change this expectation, first ensure that there is a newline after the opening
+  paren/bracket/brace.
+- ET121 (flake8-tabs) *unexpected number of tabs at start of definition line*
+
+#### Preferred formatting
+
+- Break after the parenthesis, putting the first parameter on a new line.
+- Double indent the params to avoid ET121
+
+```python
+# method with many parameters
+# start params on new line to avoid aligning with parenthesis.
+def foo(
+		arg1,  # double indent to avoid ET121
+		arg2
+):
+	# long expression
+	# start params on new line to avoid aligning with parenthesis.
+	if(
+		arg1 is not None
+		and arg2 is None
+	):
+		return None
+
+	values = [
+		"value1",
+		"value2",
+	]
+	return values
+
+```
+
+Note: A comment an inline comment causes an erroneous error from flake8-tabs:
+See https://gitlab.com/ntninja/flake8-tabs/issues/1
+EG:
+```python
+def foo(  # a comment here causes error ET128
+		arg1
+):
+	pass
+```
+
+#### Explanation
+
+An example of code that will trigger this lint error:

```

Comment:
This heading structure is a bit unclear. You're referring to ET121, right?

I think, rather than listing the errors this documentation is about in a list, using headings for every linter error we describe might be a better approach.

Reply:
I actually started with that, but then you need examples for each one, and it becomes overwhelming. ET128 is the most misleading error, because it tells you to change indentation when really the "formatting style" is wrong.